lib/statsd/instrument.rb

```diff
@@ -277,17 +273,25 @@ def backend
   #      http_response = StatsD.measure('HTTP.call.duration') do
   #        HTTP.get(url)
   #      end
-  def measure(key, value = nil, *metric_options, &block)
+  def measure(key, value = nil, *metric_options)
     if value.is_a?(Hash) && metric_options.empty?
       metric_options = [value]
       value = value.fetch(:value, nil)
     end
 
-    result = nil
-    value  = 1000 * StatsD::Instrument.duration { result = block.call } if block_given?
-    metric = collect_metric(hash_argument(metric_options).merge(type: :ms, name: key, value: value))
-    result = metric unless block_given?
-    result
+    if block_given?
+      result = nil
+      start = StatsD::Instrument.gettime
+      begin
+        result = yield
+      ensure
+        value = 1000 * StatsD::Instrument.gettime - start
+        collect_metric(hash_argument(metric_options).merge(type: :ms, name: key, value: value))
+        result
+      end
+    else
+      collect_metric(hash_argument(metric_options).merge(type: :ms, name: key, value: value))
+    end
```

Comment:
I think you can still use the old duration method. 

``` ruby
value = 1000 * StatsD::Instrument.duration do
  begin
    result = yield
  ensure
    ...
  end
end
```

Reply:
The problem is that we need to have the duration by the time we reach the `ensure` block in order to deliver the metric with the actual length of the call. I dont think this would work :(